src/bcccp/tickets/season/SeasonTicketDAO.java

```diff
@@ -6,46 +6,51 @@
 
 package bcccp.tickets.season;
 
-import java.util.ArrayList;
-import java.util.List;
+import java.util.HashMap;
+import java.util.Map;
 
 import bcccp.tickets.season.ISeasonTicket;
 import bcccp.tickets.season.IUsageRecordFactory;
 
 
 public class SeasonTicketDAO implements ISeasonTicketDAO {
 
-	private IUsageRecordFactory factory;
-	private List<ISeasonTicket> seasonTicketList = new ArrayList<ISeasonTicket>();
+	private IUsageRecordFactory usageRecordFactory;
+	private Map<String,ISeasonTicket> currentTickets; //create a map with ticketId and ticket
```

Comment:
change to Map for better tracking of ticket, ticket can be retrieved through key